.devcontainer/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM python:3.10
+FROM python:3.12
```

Comment:
@thekaveman left a good note about this in https://github.com/cal-itp/mobility-marketplace/pull/746#discussion_r2632942442

i'm fine using this repo as a testbed for 3.14, but only if everyone else agrees it adds value.